Lib/asyncio/events.py

```diff
@@ -72,6 +72,24 @@ def _format_callback_source(func, args):
     return func_repr
 
 
+def extract_stack(f=None, limit=None):
+    """Replacement for traceback.extract_stack() that only does the
+    necessary work for asyncio debug mode.
+    """
+    if f is None:
+        f = sys._getframe().f_back
+    if limit is None:
+        # This is heuristically the max number of entries we're gonna pop + 1
+        # (we're only interested in displaying the top of stack)
+        # If this is too small, tests will fail.
+        limit = 4
```

Comment:
Can we bump this to 10?  It's a debug mode after all, and seeing only 2-3 lines in traceback is sometimes not enough.

Reply:
The traceback is never displayed. It's only used for `Future.__repr__`, which prints the line at which the future was created (i.e. the top-of-stack).

Reply:
IIRC `Task.__del__` and `CoroWrapper` pass it to the loop's logger to print some debug info.  In which case the full traceback would be printed to the user, no?

Reply:
Oh, yes, you're right. I agree with bumping to 10 then.

Reply:
Can you add it as a constant to `asyncio/constants.py`?  Also, we need to update `loop.default_exception_handler()` method to say that we only display 10 topmost frames.